CHANGELOG.md

```diff
@@ -4,6 +4,151 @@ All notable changes to Router will be documented in this file.
+
+A bug in batch manipulation meant that the last element in a batch was treated as the first element. Ordering should be maintained and there is now an updated unit test to ensure this.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/4144
+
+### Port to apollo-compiler 1.0 beta ([PR #4038](https://github.com/apollographql/router/pull/4038))
+
+Version 1.0 is a near-complete rewrite of `apollo-compiler`.
+Using it in the Router unblocks a lot of upcoming work.
+
+As a more immediate benefit, some serialization-related bugs including 
+[Issue #3541](https://github.com/apollographql/router/issues/3541) are fixed.
+The representation of GraphQL documents in previous compiler versions was immutable.
+When modifying a query (such as to remove `@authenticated` fields from an unauthenticated request)
+the Router would build a new data structure with `apollo-encoder`, serialize it, and reparse it.
+`apollo-compiler`` 1.0 allows mutating a document in-place, skipping the serialization step entirely.
+
+By [@SimonSapin](https://github.com/SimonSapin) in https://github.com/apollographql/router/pull/4038
+
+### Propagate multi-value headers to subgraphs ([Issue #4153](https://github.com/apollographql/router/issues/4153))
+
+Use `HeaderMap.append` instead of `insert` to prevent erasing previous values when using multiple headers with the same name.
+
+By [@nmoutschen](https://github.com/nmoutschen) in https://github.com/apollographql/router/pull/4154
+
+## 📃 Configuration
+
+### Adds a `poll_interval` option to the JWKS endpoint in the Authentication plugin ([Issue #4185](https://github.com/apollographql/router/issues/4185))
+
+In order to avoid rate limiting concerns on JWKS endpoints, this introduces a new `poll_interval` configuration option to set the interval per each JWKS URL. The configuration option takes in a human-readable duration (e.g. `60s` or `1minute 30s`). Example that sets interval to 30 seconds: 
+
+```yml
+authentication:
+  router:
+    jwt:
+      jwks:
+        - url: https://dev-zzp5enui.us.auth0.com/.well-known/jwks.json
+          poll_interval: 30s
+```
+
+By [@lleadbet](https://github.com/lleadbet) in https://github.com/apollographql/router/pull/4212
+
+### Configuration: allow custom health check endpoint ([Issue #2938](https://github.com/apollographql/router/issues/2938))
+
+Adds a configuration option for custom health check endpoints, `health_check.path`, with `/health` as the default value.
+
+
+
+By [@aaronArinder](https://github.com/aaronArinder) in https://github.com/apollographql/router/pull/4145
+
+### Configurable JWKS Poll Interval ([Issue #4185](https://github.com/apollographql/router/issues/4185))
+
+The poll interval was previously hardcoded to 60 seconds. It is still the default now, but can be configured through the new `poll_interval` configuration option under each JWKS entry to avoid becoming rate-limited per endpoint.
```

Comment:
```suggestion
The poll interval was previously hardcoded to 60 seconds. That remains the default, but can now be configured through the new `poll_interval` configuration option under each JWKS entry to avoid becoming rate-limited per endpoint.
```